tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,321 @@
-    sessionFile = path.join(nodePath, 'token');
-    await fs.promises.writeFile(passwordFile, 'password');
-    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
-      password: 'password',
-      nodePath: nodePath,
-      logger: logger,
+    // Invalidate all active sessions
+    await testBinUtils.pkStdio(
+      ['agent', 'lock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+  });
+  afterEach(async () => {
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
     });
-    await polykeyAgent.vaultManager.createVault(vaultName);
   });
-
-  afterAll(async () => {
-    await polykeyAgent.stop();
-    await polykeyAgent.destroy();
-    await fs.promises.rmdir(dataDir, { recursive: true });
+  test('processes should store session token in session file', async () => {
+    // Run command to set token
+    let exitCode, stdout;
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    ));
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
+    await sleep(1100);
+    // Try again without password
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+      },
+      global.binAgentDir,
+    ));
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
   });
-
-  test('Process should store session token in session file', async () => {
-    // Agent has not been unlocked yet
-    const result = await testUtils.pkStdio(
-      command,
-      { PK_PASSWORD: 'password' },
-      dataDir,
-    );
-    expect(result.exitCode).toBe(0);
-    expect(result.stdout).toContain(vaultName);
-
-    const buff = await fs.promises.readFile(sessionFile);
-    const newToken = buff.toString() as SessionToken;
-    expect(
-      async () => await polykeyAgent.sessionManager.verifyToken(newToken),
-    ).not.toThrow();
+  test('processes should refresh the session token', async () => {
+    const session = await Session.createSession({
+      sessionTokenPath,
+      fs,
+      logger,
+    });
+    // Generate new token
+    await testBinUtils.pkStdio(
+      ['agent', 'unlock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+    await sleep(1100);
+    const token1 = await session.readToken();
+    // New command should refresh token
+    const { exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+      },
+      global.binAgentDir,
+    );
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
+    await sleep(1100);
```

Comment:
There is no need to sleep. The token should always be refreshed by the time the command finishes. Token refreshing is done on the client side and therefore it is not like `pk agent stop`.

Reply:
All sleeps removed https://github.com/MatrixAI/js-polykey/pull/296#discussion_r765381558